R/repr_matrix_df.r

```diff
@@ -140,24 +156,35 @@ repr_latex.matrix <- function(obj, ..., colspec = getOption('repr.matrix.latex.c
 	cols <- paste0(paste(rep(colspec$col, ncol(obj)), collapse = ''), colspec$end)
 	if (!is.null(rownames(obj)))
 		cols <- paste0(colspec$row.head, cols)
-	
+	obj <- latex.escape.names(obj)
+
+	# Using apply here will convert a data.frame to matrix, as well as escaping
+	# any columns with LaTeX specials, but only go through the hassle if there are
+	# actually things to escape.
+	if (any(apply(obj, 2L, any.latex.specials))) {
+		obj_rownames <- rownames(obj)
+		obj <- apply(obj, 2L, latex.escape.vec)
+		# If obj only has one row, apply will collapse it to a vector.
+		# That's a pain, so we'll recast it as a matrix.
+		if (is.null(dim(obj)))
```

Comment:
can you try to do `obj[] <- apply(obj, 2L, latex.escape.vec)`? i’m relatively sure this works!